apps/schedules/lib/hours_of_operation.ex

```diff
@@ -196,31 +220,94 @@ defmodule Schedules.HoursOfOperation do
   defp time(%{"departure_time" => nil, "arrival_time" => time}), do: time
   defp time(%{"departure_time" => time}), do: time
 
-  defp departure(%JsonApi{data: data}) do
-    {:ok, departure(data)}
+  # Grabs the head sign data from the stops,
+  # Letting us know the terminal stops for the trip
+  defp get_terminal_stops(data) do
+    data
+    |> Enum.map(fn data ->
+      trips = Map.get(data.relationships, "trip")
+      trip = List.first(trips)
+      Map.get(trip.attributes, "headsign")
+    end)
+    |> Enum.uniq()
   end
 
-  defp departure({:error, [%JsonApi.Error{code: "no_service"}]}) do
+  defp get_headsigns(%JsonApi{data: data_0}, %JsonApi{data: data_1}, :rapid_transit) do
+    headsign_0 = get_terminal_stops(data_0)
+    headsign_1 = get_terminal_stops(data_1)
+
+    {:ok, Enum.concat(headsign_0, headsign_1)}
+  end
+
+  defp get_headsigns(%JsonApi{data: _data_0}, %JsonApi{data: _data_1}, _) do
+    {:ok, []}
+  end
+
+  defp get_headsigns({:error, _} = error, _, _) do
+    error
+  end
+
+  defp is_terminus?(stop_name, headsigns) do
+    Enum.member?(headsigns, stop_name)
+  end
```

Comment:
I wonder if there's ever a case where the stop name at the last stop isn't exactly going to match a headsign. I'd hope not...

Reply:
I sure hope not as well.